### ISSUE TITLE
Public notebooks shouldn't be modified directly by other users

### ISSUE THREAD

**Issue by emanueles** [Microsoft/AzureNotebooks#191]
Please consider the template below to file your issue. 
### Azure Notebooks environment 
notebooks.azure.com
### Description
I noticed my notebooks in my public libraries are edited by other users without my permission.

### Repro Steps
Unfortunately, I don't know how this is happening. I just shared a few libraries for students to clone and modify on their own, but sometimes they forget to clone the notebooks before and are able to edit the notebook directly in my library.

### Usage Scenario (e.g. teaching a computing course at university foo)
I am sharing some libraries for exercises and homework assignments on my account with my students. Although I told them to clone the libraries to their accounts, somehow they are able to edit the notebooks directly in my library. Is this the expected behavior? Is there a way to prevent this?

### Recommended Metadata
The Jupyter Kernel (Python2, Python3, R, N/A): Python3
Occurrence Time: From September, 23 to September, 28.
Web Browser: Chrome
LibraryID and Notebook Name (if applicable): https://notebooks.azure.com/emanueles/libraries/metodos

Logs (!cat .nb.log Or: New/Terminal, $cat .nb.log) - attach files if applicable

**Comment by crwilcox** [comment]
@emanueles, we are looking into this now. If we need more information we will reach out to you.

**Comment by crwilcox** [comment]
@emanueles , I was able to find an odd case, which I think is what you are encountering. I wanted to let you know that, from what I can tell, your actual library is not being modified.

However, the preview is not accurate and is actually showing your student's preview. Let me explain more thoroughly.

If a student clones a library, edits their copy, and previews their copy, the main notebook (yours) sees their preview. That shouldn't be happening. This is a bug. The good news is they aren't able to edit your files.

I am working on resolving this issue now.

Thank you for reporting this.

**Comment by crwilcox** [comment]
@emanueles , I have deployed a hotfix and validated it. This behavior is now correct.

Also, just to mention again, your data was never edited by students. The preview attached to your library however was. It seems we introduced a bug where children of clones and their parents would, in certain cases, share the same preview. So when one changed, the preview would change and be displayed for both files.

Thanks again for reporting this!
-Chris
